### PR TITLE
Don't pull the whole world

### PR DESCRIPTION
**Pull request by andremedeiros**
r. @Shopify/sigint

### CODE REVIEW

File: lib/doggy/cli/pull.rb [hunk 1]
```diff
@@ -2,37 +2,61 @@
...
     end
 
   private
 
-    def should_pull?(resource)
-      !@options.empty? || @options[resource] || @ids.any?
-    end
+    def pull_by_id(id)
+      local_resource = @local_resources.find { |l| l.id == id }
+      if !local_resource then
```
Comment (kvs) [inline lib/doggy/cli/pull.rb]: then?!? :trollface:

File: lib/doggy/model.rb [hunk 2]
```diff
@@ -56,6 +64,14 @@ def all
         Parallel.map(ids) { |id| find(id) }
       end
 
+      def all_local_resources
+        @resources ||= begin
```
Comment (kvs) [inline lib/doggy/model.rb]: Useless use of begin..end

**Comment by kvs** [comment]
:shirt: less RT plus p

**Comment by andremedeiros** [comment]
SHITP?

**Pull request merged**